Modules/booking/Controller/BookingauthorisationsController.php

```diff
@@ -21,31 +21,52 @@
  */
 class BookingauthorisationsController extends CoresecureController {
 
-    public function indexAction($id_space, $id) {
-
+    public function indexAction($id_space, $id_user) {
         $this->checkSpaceAdmin($id_space, $_SESSION["id_user"]);
         $lang = $this->getLanguage();
 
-        // get all the resources
-        $modelResources = new ReCategory();
-        $resources = $modelResources->getBySpace($id_space);
-
         $modelSpace = new CoreSpace();
         $space = $modelSpace->getSpace($id_space);
 
-        // user name
-        $modelUser = new CoreUser();
-        $userName = $modelUser->getUserFUllName($id);
+        $form = $this->generateBkAuthAddForm($id_space, $id_user, "bookingauthorisations");
+        $generatedBkAuth = $this->generateBkAuthTable($id_space, $id_user, "bookingauthorisations");
+        $tableHtml = $generatedBkAuth['bkTableHtml'];
+        $bkAuthData = $generatedBkAuth['data'];
 
+        if ($form->check()) {
+            $this->validateBkAuthAddForm(
+                $id_space,
+                $id_user,
+                $form->getParameter("resource") /* stands for category id */,
+                $form->getParameter("visa_id"),
+                $form->getParameter("date")
+            );
+        }
+
+        return $this->render(array(
+            "lang" => $lang,
+            "id_space" => $id_space,
+            'tableHtml' => $tableHtml,
+            "formHtml" => $form->getHtml($lang),
+            'space' => $space,
+            'data' => $bkAuthData
+        ));
+    }
+
+    public function generateBkAuthTable($id_space, $id_user, $controller) {
+        $this->checkSpaceAdmin($id_space, $_SESSION["id_user"]);
```

Comment:
Dup with method calling it, and should get lang as param to avoid recalculating it